packages/streams/tcp.js

```diff
@@ -42,6 +42,15 @@ function TcpStream (options) {
 require('util').inherits(TcpStream, Transform)
 
 TcpStream.prototype.pipe = function (pipeTo) {
+  if ( this.options.outEvent ) {
+    const that = this
```

Comment:
If the tcp stream is not connected yet / is disconnected the event handler will throw. Only write if `that.tcpStream` is defined? And make sure you unset it if we lose the connection.

Reply:
That is checked below in the event handler. So I think I just need to unset it?

Reply:
👍